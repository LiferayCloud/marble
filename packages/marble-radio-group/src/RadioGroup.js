'use strict';

import {core} from 'metal';
import Component from 'metal-component';
import Soy from 'metal-soy';

import templates from './RadioGroup.soy.js';

/**
 * RadioGroup component.
 */
class RadioGroup extends Component {
  created() {

  }

  disposed() {

  }
}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
RadioGroup.STATE = {
  /**
   * ID to be applied to the element.
   * @type {!Object}
   */
  id: {
    validator: core.isString,
    value: 'radio-group'
  },
};

Soy.register(RadioGroup, templates);

export {RadioGroup};
export default RadioGroup;
