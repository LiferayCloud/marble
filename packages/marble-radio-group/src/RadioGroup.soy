{namespace RadioGroup}

/**
 * This renders the component's whole content.
 */
{template .render}
  <div class="radio-group">Hello World</div>
{/template}
