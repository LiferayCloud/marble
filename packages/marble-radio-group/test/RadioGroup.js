'use strict';

import RadioGroup from '../src/RadioGroup';

let radioGroup;

describe('RadioGroup', function() {
  afterEach(() => {
    if (radioGroup) {
      radioGroup.dispose();
    }
  });

  it('should generate the default markup', () => {
    radioGroup = new RadioGroup();

    expect(radioGroup).toMatchSnapshot();
  });
});
