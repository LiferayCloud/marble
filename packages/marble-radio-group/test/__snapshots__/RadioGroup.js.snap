// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should generate the default markup 1`] = `<div class="radio-group">Hello World</div>`;
